domains/transfer/Transfer.API/API.IntegrationTests/Controllers/CvrControllerTests.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Net;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using API.Cvr.Dtos;
+using API.IntegrationTests.Factories;
+using API.IntegrationTests.Mocks;
+using FluentAssertions;
+using Xunit;
+
+namespace API.IntegrationTests.Controllers;
+
+public class CvrControllerTests : IClassFixture<TransferAgreementsApiWebApplicationFactory>,
+    IClassFixture<CvrWireMock>
+{
+    private readonly TransferAgreementsApiWebApplicationFactory factory;
+    private readonly CvrWireMock dataSyncWireMock;
+
+    public CvrControllerTests(TransferAgreementsApiWebApplicationFactory factory,
+        CvrWireMock dataSyncWireMock)
+    {
+        this.factory = factory;
+        this.dataSyncWireMock = dataSyncWireMock;
+        factory.WalletUrl = "UnusedWalletUrl";
+        factory.CvrBaseUrl = dataSyncWireMock.Url;
+    }
+
+    [Fact]
+    public async Task GetCvrCompany_WhenCorrectCvrNumber_ShouldReturnCvrInfo()
+    {
+        var cvrNumber = "10150817";
+        dataSyncWireMock.SetupCvrResponse();
+
+        var client = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString());
+
+        var response = await client.GetFromJsonAsync<CvrCompanyDto>($"api/cvr/{cvrNumber}");
+
+        response.Should().NotBeNull();
+        response.CompanyCvr.Should().Be(cvrNumber);
+        response.Address.Should().NotBeNull();
+        response.Address.Kommune.Should().NotBeNull();
```

Comment:
Add assertions on the actual response. You could use Verify.xUnit for making this easy.

Reply:
corrected